### ISSUE TITLE
Setting up a dev. environment

### ISSUE THREAD

**Issue by michaelaye** [USGS-Astrogeology/plio#83]
```bash
(py36) └─❱❱❱ pip install -e .                                     +8980 11:05 ❰─┘
Obtaining file:///Users/klay6683/src/plio
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/Users/klay6683/src/plio/setup.py", line 3, in <module>
        import plio
      File "/Users/klay6683/src/plio/plio/__init__.py", line 18, in <module>
        from . import io
      File "/Users/klay6683/src/plio/plio/io/__init__.py", line 25, in <module>
        from . import io_controlnetwork
      File "/Users/klay6683/src/plio/plio/io/io_controlnetwork.py", line 7, in <module>
        from plio.io import ControlNetFileV0002_pb2 as cnf
      File "/Users/klay6683/src/plio/plio/io/ControlNetFileV0002_pb2.py", line 6, in <module>
        from google.protobuf import descriptor as _descriptor
    ModuleNotFoundError: No module named 'google'
```
The install_dependencies also don't mention `google` ?

```python
install_requires=[
            'gdal',
            'numpy',
            'pvl',
            'protobuf',
            'h5py',
            'pandas',
            'sqlalchemy',
            'pyyaml',
            'networkx',
            'affine',
            'scipy'],
```

but `protobuf`. Related?

System: macOS 10.13.6

**Comment by jlaura** [comment]
@michaelaye We do not support a pip install, only a conda install. This is due to difficulty with protobuf and gdal installations.

**Comment by michaelaye** [comment]
So how would I install it then if I want to add something?

**Comment by jlaura** [comment]
https://github.com/USGS-Astrogeology/plio#installation

`conda install -c usgs-astrogeology plio`

**Comment by michaelaye** [comment]
wow, okay, never saw a dev install via conda. Why would I edit source code directly inside the site-packages folder, that's immensely dirty. Guess I should add my image drivers to planetaryimage then.

**Comment by jlaura** [comment]
@michaelaye I am really confused about what you are trying to do?  Are you looking to install plio to use or are you looking to develop for plio?

**Comment by michaelaye** [comment]
I did say "if I want to add something"? Maybe I should have added "to plio"? 
I would have hoped that the fact that I'm teaching others how to properly work with conda would have elevated me over the other interpretation of my sentence of "if I want to add something <to my python install thingie>" but I guess I failed. (Wasn't it me that taught you about environment-specific conda channel configurations? ;)
In other words, yes, I see an urgent need to add more planetary data formats and am currently scouting where this should be happening, so I'm looking for a dev install.

**Comment by jlaura** [comment]
I think we are different issues? This one is about pip installing plio. I would never pip install something that I was developing on. I would run a ` python setup.py develop` in the source tree.  For developing plio, we would run `conda env create -f environment.yml` in the source tree. This assumes that conda-forge is at the top of your conda channel list. From there, you can `setup.py develop` plio and have a stable development environment.

I am slowly porting our projects to have the travis env also use the environment.yml to avoid having travis and environment.yml not be in sync.

**Comment by jlaura** [comment]
I should add that I am hoping to have time to get everything into just setup.py and then have the meta.yml (in the recipe) and the environment.yml all pull from a single place. The duplication in where dependencies are being specified is a maintenance headache.

**Comment by michaelaye** [comment]
`pip install -e .` (note the `-e` which is equivalent to `develop`) is better than `python setup.py develop` because pip keeps track of more meta-data to the install, specifically, a cleaner removal.
You also could just add `-c conda-forge` to your environment setup to ensure that conda-forge is preferred. Thanks, I'll try this route.

**Comment by jlaura** [comment]
:+1: on the `-e`. I think though that protobuf will be an issue for the setup.py. I need to try the pip 3.6.x protobuf to see what is up there. I am going to repoen this and retitle setting up a dev environment so that we can close the loop accurately.

**Comment by michaelaye** [comment]
Hm, I see the double-load of work keeping things checked at 2 locations. I'm guessing that the conda environment route is the way of the future, what do you think? Or will pip exist in parallel forever? I myself haven't used the environment yaml files route that much, but if you were to restrict the dev install to the conda envs only, I think that would totally be justifiable. Unless you would specifically want to include pip-only devs.

**Comment by jlaura** [comment]
I just opened #84 to get this synced in at least one more place. I would love for setup.py to be the one and only palce this is ever defined. Then treat environment.yml and meta.yml and jinja templates and inject from there.

In an ideal world I would prefer to support pip/wheel and conda. My issue currently is with gdal and the way that the binaries (C++) are or are not installed via pip. If I could confidently say that `pip install plio` would get GDAL, I would be sold.

**Comment by jlaura** [comment]
@michaelaye I just merged #84, so the environment.yml is testing good fot setting up a dev. environment, cloning plio, and then running pytest. The merge will take a bit to build in the CI and update anaconda.org, but that should not impact you being able to see if this works on your side.

**Comment by jlaura** [comment]
:tada: Thanks! 
Are you happy with closing this issue?  Looks like we got it resolved. Also, appreciate any feedback / implementation on growing our driver base.
